lib/src/pubspec.dart

```diff
@@ -1,21 +1,36 @@
+import 'dart:collection';
+
+import 'package:yaml/yaml.dart' as yaml;
+
+import 'utils.dart';
+
 class Pubspec {
   final Map<String, dynamic> _data;
 
+  Set<String> _dependentSdks;
+
   Pubspec(this._data);
 
+  factory Pubspec.parseFromDir(String packageDir) {
+    String content = getPubspecContent(packageDir);
+    if (content == null) {
+      throw new Exception("Couldn't find a pubspec.yaml in $packageDir.");
+    }
+    return new Pubspec(yaml.loadYaml(content));
+  }
+
   Map<String, dynamic> get dependencies => _data['dependencies'];
+  Map<String, dynamic> get devDependencies => _data['dev_dependencies'];
 
   bool get hasFlutterKey => _data.containsKey('flutter');
   bool get hasFlutterPluginKey =>
       hasFlutterKey &&
       _data['flutter'] is Map &&
       _data['flutter']['plugin'] != null;
 
-  bool get dependsOnFlutterSdk {
-    if (dependencies == null) return false;
-    return dependencies.values
-        .any((value) => value is Map && value['sdk'] == 'flutter');
-  }
+  bool get dependsOnFlutterSdk =>
```

Comment:
Couldn't this just be `dependentSdks.contains('flutter')`?

This we can get rid of the `_dependsOnFlutterSdk` helper? Seems a bit redundant...